lib/CSkillHandler.h

```diff
@@ -87,6 +88,10 @@ class DLL_LINKAGE CSkill : public Skill
 	friend class CSkillHandler;
 	friend DLL_LINKAGE std::ostream & operator<<(std::ostream & out, const CSkill & skill);
 	friend DLL_LINKAGE std::ostream & operator<<(std::ostream & out, const CSkill::LevelInfo & info);
+private:
+	//Do not serialize, used only on level up heroes
+	bool obligatoryMajor;
+	bool obligatoryMinor;

```

Comment:
Please serialize. We serialize all properties of all objects, even if loaded on startup

Reply:
@IvanSavenko okay, but it will break save format in more places.

Reply:
You're breaking it already, so see no reason to use such workaround.

Whether you make this change or not the result it the same - old saves won't load.

Reply:
Done serialization. But I do not understand why we should serialize it? it is not useful to save and load it from savegames.

Reply:
I kind of agree, not much point behind saving library handlers. We require same set of mods anyway. But to keep things consistent better to serialize everything for now.